collagraph/renderers/pyside_renderer.py

```diff
@@ -99,6 +100,14 @@ def not_implemented(self, *args, **kwargs):
 class PySideRenderer(Renderer):
     """PySide6 renderer."""
 
+    def register(self, type_name, custom_type):
+        # Check that the custom type is a subclass of QWidget.
+        # This ensures that the custom widget can be properly wrapped
+        # and will get the `insert`, `remove` and `set_attribute`
+        # methods.
+        assert QWidget in custom_type.__mro__
```

Comment:
In general, try to use assert only in test suite code. Asserts are skipped by the interpreter when running in optimized mode, as for example pyinstaller will do!